### PR TITLE
chores: add vendor update job

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind testing 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -101,7 +101,7 @@ help: ## Display this help.
 ## --------------------------------------
 
 .PHONY: all
-all: $(BIN_DIR)/azure-cloud-controller-manager $(BIN_DIR)/azure-cloud-node-manager $(BIN_DIR)/azure-cloud-node-manager.exe ## Build binaries for the project.
+all: $(BIN_DIR)/azure-cloud-controller-manager $(BIN_DIR)/azure-cloud-node-manager $(BIN_DIR)/azure-cloud-node-manager.exe $(BIN_DIR)/azure-acr-credential-provider $(BIN_DIR)/azure-acr-credential-provider.exe ## Build binaries for the project.
```
Comment (lzhecheng) [inline Makefile:104]: Is this change related to vendor license? If not, how about creating a separate PR for it? Maybe it needs backport as well.

Comment (MartinForReal) [reply Makefile:104]: all changes should be back-ported.
